appcontainer/ThemeModel.js

```
@@ -32,10 +32,25 @@ export class ThemeModel extends HoistModel {
         classList.toggle('xh-dark', value);
         classList.toggle('bp3-dark', value);
         this.darkTheme = value;
-        XH.setPref('xhTheme', value ? 'dark' : 'light');
+
+    }
+
+    @action
+    setThemePref(value) {
+        if (value === 'system') {
```

think a switch with three clauses would actually be clearer at this point.

---

also should throw if presented with a unknown value

---

also should throw if presented with a unknown value